web3/_utils/module_testing/eth_module.py

```diff
@@ -1547,6 +1547,251 @@ async def test_async_eth_sign_ens_names(
+        replace_txn = await async_w3.eth.get_transaction(replace_txn_hash)
+
+        assert is_same_address(
+            replace_txn["from"], cast(ChecksumAddress, txn_params["from"])
+        )
+        assert is_same_address(
+            replace_txn["to"], cast(ChecksumAddress, txn_params["to"])
+        )
+        assert replace_txn["value"] == 1
+        assert replace_txn["gas"] == 21000
+        assert replace_txn["maxFeePerGas"] == three_gwei_in_wei
+        assert replace_txn["maxPriorityFeePerGas"] == two_gwei_in_wei
+
+    @pytest.mark.asyncio
+    async def test_async_eth_replace_transaction_underpriced(
+        self, async_w3: "AsyncWeb3", unlocked_account_dual_type: ChecksumAddress
+    ) -> None:
+        txn_params: TxParams = {
+            "from": unlocked_account_dual_type,
+            "to": unlocked_account_dual_type,
+            "value": Wei(1),
+            "gas": 21000,
+            "maxFeePerGas": async_w3.to_wei(3, "gwei"),
+            "maxPriorityFeePerGas": async_w3.to_wei(2, "gwei"),
+        }
+        txn_hash = await async_w3.eth.send_transaction(txn_params)
+
+        one_gwei_in_wei = async_w3.to_wei(1, "gwei")
+        txn_params["maxFeePerGas"] = one_gwei_in_wei
+        txn_params["maxPriorityFeePerGas"] = one_gwei_in_wei
+
+        with pytest.raises(ValueError, match="replacement transaction underpriced"):
+            await async_w3.eth.replace_transaction(txn_hash, txn_params)
+
+    @pytest.mark.asyncio
+    async def test_async_eth_replace_transaction_non_existing_transaction(
+        self, async_w3: "AsyncWeb3", unlocked_account_dual_type: ChecksumAddress
+    ) -> None:
+        txn_params: TxParams = {
+            "from": unlocked_account_dual_type,
+            "to": unlocked_account_dual_type,
+            "value": Wei(1),
+            "gas": 21000,
+            "maxFeePerGas": async_w3.to_wei(3, "gwei"),
+            "maxPriorityFeePerGas": async_w3.to_wei(1, "gwei"),
+        }
+        with pytest.raises(TransactionNotFound):
+            await async_w3.eth.replace_transaction(
+                HexStr(
+                    "0x98e8cc09b311583c5079fa600f6c2a3bea8611af168c52e4b60b5b243a441997"
+                ),
+                txn_params,
+            )
+
+    @pytest.mark.asyncio
+    @pytest.mark.xfail(reason="AsyncGethMiner is missing.")
+    async def test_async_eth_replace_transaction_already_mined(
+        self, async_w3: "AsyncWeb3", unlocked_account_dual_type: ChecksumAddress
+    ) -> None:
+        txn_params: TxParams = {
+            "from": unlocked_account_dual_type,
+            "to": unlocked_account_dual_type,
+            "value": Wei(1),
+            "gas": 21000,
+            "maxFeePerGas": async_w3.to_wei(2, "gwei"),
+            "maxPriorityFeePerGas": async_w3.to_wei(1, "gwei"),
+        }
+        txn_hash = await async_w3.eth.send_transaction(txn_params)
+        try:
+            async_w3.geth.miner.start()  # type: ignore
```

Comment:
The AsyncGethMiner is not implemented, so I mark this test as an expected failure.

Reply:
The go-ethereum [`miner`](https://geth.ethereum.org/docs/interacting-with-geth/rpc/ns-miner) namespace is now deprecated, but I can help implement it if needed, just like the other async Geth namespaces.